### PR TITLE
Prizes in neighbourhood calculation

### PR DESCRIPTION
**Pull request by N-Wouda**
Related to #215. This PR adds prize data to the neighbourhood proximity calculation.

<details>

**Notes**:

- It is essential that you add a test when making code changes.
  This keeps the code coverage level up, and helps ensure the changes work as intended.
- When fixing a bug, you must add a test that would produce the bug in the master branch, and then show that it is fixed with the new code. 
- New code additions must be well formatted. Changes should pass the pre-commit workflow, which you can set up locally using [pre-commit](https://pre-commit.com/#intro). 
- Docstring additions must render correctly, including escapes and LaTeX.

</details>

### CODE REVIEW

**Comment by N-Wouda** [comment]
I motivate this change with the following benchmark results 😆:
image

**Comment by N-Wouda** [comment]
(we might want to look at $k$ for PC-VRPTW as well)

File: pyvrp/educate/tests/test_neighbourhood.py [hunk 1]
```diff
@@ -163,3 +163,19 @@ def test_more_neighbours_than_instance_size():
...
+    data = read("data/p06-2-50.vrp", round_func="dimacs")
+    params = NeighbourhoodParams(0, 0, nb_granular=10)
+    neighbours = compute_neighbours(data, params)
+
+    # We compare the number of times clients 20 and 36 are in other clients'
+    # neighbourhoods. Client 20 is at location (57, 58), client 36 at (63, 69).
+    # They're fairly close to each other, in one corner of the plane. The
+    # biggest difference is in prizes: client 20 has a prize of 33, whereas
+    # client 36 only yields 8. As a consequence, 36 should be in many fewer
+    # neigbhourhoods than 20.
```
Comment (leonlan) [inline pyvrp/educate/tests/test_neighbourhood.py:178]:
```suggestion
    # neighbourhoods than 20.
```

Comment (N-Wouda) [reply pyvrp/educate/tests/test_neighbourhood.py]: Spelling is hard 👍

**Review by leonlan — APPROVED** [review]

**Pull request merged**